src/tenable/test/test_pipeline.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+import subprocess
+from pathlib import Path
+from mariadb import mariadb
+
+from src.tenable.pipeline import *
+from src.tenable.test.conftest import TEST_ASSET_EXPORT_DIR
+from src.tenable.test.conftest import TEST_ASSET_EXPORT_DIR, TEST_VULN_EXPORT_DIR, TEST_CONN_PARAMS, TEST_CONN_PARAMS_DB, MIGRATION
+from src.tenable.load_assets import load_tenable_assets
+
+@pytest.fixture(autouse=True)
+def fake_filesystem(fs):
+    yield fs
+
+@pytest.fixture(autouse=True)
+def create_testdb(fs, mocker, asset_export_status):
+    conn = mariadb.connect(**TEST_CONN_PARAMS)
+    conn.begin()
+    cursor = conn.cursor()
+    cursor.execute("CREATE DATABASE IF NOT EXISTS testdb;")
+    conn.commit()
+    subprocess.run(MIGRATION)
+    conn = mariadb.connect(**TEST_CONN_PARAMS_DB)
+    fs.add_real_file(TEST_ASSET_EXPORT_DIR / "0_TEST_1.json")  # Test asset data is loaded early so that load_vulnerabilities can reference asset_uuids as a foreign key
+    mocker.patch("src.tenable.load_assets.ASSET_EXPORT_DIR", TEST_ASSET_EXPORT_DIR)
+    mocker.patch("src.tenable.load_assets.CONN_PARAMS", TEST_CONN_PARAMS_DB)
+    load_tenable_assets(asset_export_status)
+    conn.commit()
+    yield
+    conn.begin()
+    cursor = conn.cursor()
+    cursor.execute("DROP DATABASE IF EXISTS testdb;")
+    conn.commit()
+
+
+def test_tenable_success(fs, mocker):
+    mocker.patch("src.tenable.load_vulnerabilities.CONN_PARAMS", TEST_CONN_PARAMS_DB)
```

Comment:
avoid monkey patching imports - i see that you removed the `db` param in the functions that require the database. you can use that parameter to pass the test db instance.